### ISSUE TITLE
Value equality checking using 'is'

### ISSUE THREAD

**Issue by jdmunro** [facebook/immutable-js#498]
Could someone help me understand why the following assertions work:

```javascript
var name = "Tim Cook";

var list1 = Immutable.fromJS([{name: "Name: Tim Cook"}]);
var list2 = Immutable.fromJS([{name: "Name: " + name}]);

var map1 = Immutable.fromJS({name: "Name: Tim Cook"});
var map2 = Immutable.fromJS({name: "Name: " + name});

expect(Immutable.is(list1, list2)).toBe(true);
expect(Immutable.is(map1, map2)).toBe(true);
```

But in the following example, the list assertion fails because it returns false. For the maps comparison it returns true as expected though.

```javascript
var name = "Tim Cook";

var list1 = Immutable.List([{name: "Name: Tim Cook"}]);
var list2 = Immutable.List([{name: "Name: " + name}]);

var map1 = Immutable.Map({name: "Name: Tim Cook"});
var map2 = Immutable.Map({name: "Name: " + name});

expect(Immutable.is(list1, list2)).toBe(true);
expect(Immutable.is(map1, map2)).toBe(true);
```

Am I using the API incorrectly?

**Comment by leebyron** [comment]
The collection constructor functions List and Map accept any kind of children, so in your second example you have a list of references to JavaScript objects. Since every JavaScript object is unique and not equal ({} !== {}) the two lists of JavaScript object will also not be equal. 

fromJS, on the other hand, converts it's input deeply - replacing JavaScript array with List and JavaScript object with Map at every level of the input. So in that first example you have a List of Map, both of which use value semantics and are equivalent. 

If you're ever not sure what the shape of your data looks like, try calling .toString() on it to see this illustrated in a pseudo-JSON form. 

To update your second case to pass, your collection construction should explicitly create a map:

List([Map({name: "Name: Tim Cook"})]);

Let me know if this doesn't answer your question.

**Comment by jdmunro** [comment]
Thanks @leebyron that explains it just fine!

**Comment by kedashoe** [comment]
@leebyron, been lurking around here for quite a while now, first post so let me say :+1: for your work here and :100: for your outstanding communication!

This issue comes up again and again in one form or another. Two of three most recently closed issues are related to it: #493 and this one. I don't see anything at a quick glance at the README that shows this behaviour. Maybe in the https://github.com/facebook/immutable-js#nested-structures section, you could add some language and an example for list and map that shows what happens (and maybe even move the section up)? I think basically what you've written here is a great explanation.

Thanks again for all your hard work
